app/models/user.rb

```diff
@@ -2,15 +2,32 @@
 class User < ApplicationRecord
   # Connects this user object to Blacklights Bookmarks.
   include Blacklight::User
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+
+  # Include devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  # remove :database_authenticatable in production, remove :validatable to integrate with Shibboleth
+  devise_modules = [:omniauthable, :rememberable, :trackable, omniauth_providers: [:shibboleth], authentication_keys: [:uid]]
+  devise_modules.prepend(:database_authenticatable) if AuthConfig.use_database_auth?
+  devise(*devise_modules)
 
   # Method added by Blacklight; Blacklight uses #to_s on your
   # user class to get a user-displayable login/identifier for
   # the account.
   def to_s
-    email
+    uid
+  end
+
+  # When a user authenticates via shibboleth, find their User object or make
+  # a new one. Populate it with data we get from shibboleth.
+  # @param [OmniAuth::AuthHash] auth
+  def self.from_omniauth(auth)
+    Rails.logger.debug "auth = #{auth.inspect}"
+    # Uncomment the debugger above to capture what a shib auth object looks like for testing
+    user = where(provider: auth.provider, uid: auth.info.uid).first_or_create
```

Comment:
Are we creating users when logging in via shibboleth? If not, we only do a hard `find_by` else error out to say `user doesn't exist` Please refer: https://github.com/emory-libraries/dlp-lux/blob/main/app/models/user.rb#L35

Reply:
This applies in the same way to the above comment.

Reply:
I was referring to lux in my comment. Please take a look at the line I have referred (which is from lux).

Reply:
I spoke earlier to @rotated8 about going further than DCE's initial Shib integration that they did in Lux, and he said that the main intent of the ticket was just to get the Shibboleth login working, and to ignore any refinement or guessing of PO requirements.

Reply:
I wasn't aware of this, thanks for letting me know. I was looking at what lux had as a final implementation for shib. My earlier comment doesn't make sense in this situation then, I apologize.

Reply:
No, no apologies needed. I should have documented the details of my conversation with Collin here before pushing work at all, so I'm the one to apologize--and I do.